File: Sistema_de_usuarios/cuenta.py
from StrongPasswordValidator import StrongPasswordValidator

class Cuenta:
    def __init__(self, nombre, contraseña, validador):
        if not validador.es_contraseña_valida(contraseña):
            raise ValueError("Contraseña inválida")
        self.user = nombre
        self.contraseña_hasheada = validador.hashear(contraseña)

    def get_nombre(self):
        return self.user

    def get_contraseña_hasheada(self):
        return self.contraseña_hasheada

File: tp1_2025_ejer/src/ejer/Cuenta.java
package ejer;

public class Cuenta {
	private String User;
	private String contraseñaHasheada;

	public Cuenta(String nombre, String contraseña, PasswordValidator validador) {
		if (nombre == null || contraseña == null)
            throw new IllegalArgumentException("Nombre y/o contraseña no pueden ser nulos");
		
		if (!validador.esContraseñaValida(contraseña))
			throw new IllegalArgumentException("Contraseña inválida");
		
		this.User = nombre;
		this.contraseñaHasheada = validador.hashear(contraseña);
	}

	public String getNombre() {
		return this.User;
	}

	public String getContraseñaHasheada() {
		return contraseñaHasheada;
	}
}
